apps/prairielearn/src/pages/instructorCopyPublicCourseInstance/instructorCopyPublicCourseInstance.ts

```diff
@@ -0,0 +1,33 @@
+import { Router } from 'express';
+import asyncHandler from 'express-async-handler';
+
+import * as error from '@prairielearn/error';
+
+import { copyCourseInstanceBetweenCourses } from '../../lib/copy-content.js';
+import { selectCourseInstanceById } from '../../models/course-instances.js';
+import { selectCourseById } from '../../models/course.js';
+
+const router = Router();
+
+router.post(
+  '/',
+  asyncHandler(async (req, res) => {
+    const courseInstance = await selectCourseInstanceById(req.body.course_instance_id);
+    if (courseInstance === null) {
+      throw new error.HttpStatusError(404, 'Not Found');
+    }
+    const course = await selectCourseById(courseInstance.course_id);
+
+    if (!courseInstance.share_source_publicly) {
+      throw new error.HttpStatusError(400, 'Copying this course instance is not permitted');
```

Comment:
This would be a 403 since it's an authorization issue, right?

Reply:
Good question, I didn't think too hard about this. 

I'm actually inclined to go with 404 because if the course instance is not publicly shared, then the public page for it 'doesn't exist' in the sense that there is no condition in which we will ever return the page. 

But I guess 404 or 403 would be fine. What do you think?